src/wp-includes/functions.php

```diff
@@ -3063,6 +3064,33 @@ function wp_get_image_mime( $file ) {
 		} else {
 			$mime = false;
 		}
+
+		if ( false !== $mime ) {
+			return $mime;
+		}
+
+		$handle = fopen( $file, 'rb' );
+		if ( false === $handle ) {
+			return false;
+		}
+
+		$magic = fread( $handle, 12 );
+		if ( false === $magic ) {
+			return false;
+		}
+
+		// Add WebP fallback detection when image library doesn't support WebP.
+		$magic = bin2hex( $magic );
+		if (
+			// RIFF.
+			( 0 === strpos( $magic, '52494646' ) ) &&
+			// WEBP.
+			( 16 === strpos( $magic, '57454250' ) )
+		) {
```

Comment:
It would be good to add a bit more information in the comment about where these values come from, e.g. a link to some supporting documentation/research. Without any "proof" anyone looking at this code later will have no idea why these values are what they are.

Reply:
Good suggestion, I'll try to track that down.

Reply:
Added a link to the reference material